third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -1459,6 +1633,49 @@ struct AtomicRMWOpConversion
     return endBlock->getArgument(0);
   }
 };
+
+struct AsyncWaitConversion : public ConvertOpToLLVMPattern<AsyncWaitOp> {
+  using ConvertOpToLLVMPattern<AsyncWaitOp>::ConvertOpToLLVMPattern;
+
+  AsyncWaitConversion(LLVMTypeConverter &converter,
+                      const AMD::TargetInfo &targetInfo,
+                      ModuleAxisInfoAnalysis &axisAnalysisPass,
+                      PatternBenefit benefit)
+      : ConvertOpToLLVMPattern<AsyncWaitOp>(converter, benefit) {}
+
+  LogicalResult
+  matchAndRewrite(AsyncWaitOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto loc = op->getLoc();
+    auto b = TritonLLVMOpBuilder(loc, rewriter);
+    rewriter.create<ROCDL::WaitcntOp>(loc, op.getNum());
```

Comment:
Can you be explicit regarding the usage of waitcnt here? Add comments regarding

* What's the semantics of waitcnt op
* How do we control waitcnt for different signals (vmem, lds, etc.)
* How is the num mapped here
* ...